### PR TITLE
Add search feature to blog page

### LINKED ISSUE #623: [NG] search plugin
**Description**

Implement the search plugin on the `layer5-ng` site.

---
**Contributor Resources**
The `layer5` repo contains two websites. The current generation and the next-generation of the layer5.io site.

If the `layer5-ng` label is absent on this issue, then this issue pertains to the current generation of the layer5.io website, which uses Jekyll and GitHub Pages. Site content is found under the [`master` branch](https://github.com/layer5io/layer5/tree/master).
- See [`layer5` contributing instructions](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md)

If the `layer5-ng` label is present on this issue, then this issue pertains to the next-generation of the layer5.io website, which uses Gatsby, Strapi, and GitHub Pages. Site content is found under the [`layer5-ng` branch](https://github.com/layer5io/layer5/tree/layer5-ng).
- See [`layer5-ng` contributing instructions](https://github.com/layer5io/layer5/blob/layer5-ng/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by kayceeDev**
Signed-off-by: Ekene Nwobodo <nwobodoe71@gmail.com>

**Description**

This PR fixes #623 

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit a1701aa9867c06bf04aae34cf635be1f8f34ffa7 at: https://623f53a990645f5d72d6fe55--layer5ng.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 3bfb9e50547e99b33bdb645c4494bc72ef53307c at: https://623f5c8ce851e35cb273f7dc--layer5ng.netlify.app

**Review by YashKamboj — COMMENTED** [review]
Please don't print the list in the search component only. If you make the search component dynamic, we can use it on many pages like blog, resources, news etc. @kayceeDev

**Review by Abhijay007 — COMMENTED** [review]
@kayceeDev it will be great if you center align search field in the tab view : ) 

image

**Comment by kayceeDev** [comment]
> @kayceeDev it will be great if you center align search field in the tab view : )
> 
> image

I will adjust the styles.

**Comment by kayceeDev** [comment]
> Please don't print the list in the search component only. If you make the search component dynamic, we can use it on many pages like blog, resources, news etc. @kayceeDev

They're reasons this method won't work. You know the search component is also filtering the state and then rendering the filtered state. This means that it only filters the state it's given and end up Saving it in the components state. There is no way to pass the state of a child to the parent.

The other alternative might be to have a general state that the search component filters and this might require redux and some other complex logic. We can have a chat about this if you think there are easy ways to achieve this.

**Comment by YashKamboj** [comment]
> They're reasons this method won't work. You know the search component is also filtering the state and then rendering the filtered state. This means that it only filters the state it's given and end up Saving it in the components state. There is no way to pass the state of a child to the parent.
> 
> The other alternative might be to have a general state that the search component filters and this might require redux and some other complex logic. We can have a chat about this if you think there are easy ways to achieve this.

Ohh!! Thats why i was also stuck with the issue 😅😅.

**Comment by kayceeDev** [comment]
> > They're reasons this method won't work. You know the search component is also filtering the state and then rendering the filtered state. This means that it only filters the state it's given and end up Saving it in the components state. There is no way to pass the state of a child to the parent.
> > The other alternative might be to have a general state that the search component filters and this might require redux and some other complex logic. We can have a chat about this if you think there are easy ways to achieve this.
> 
> Ohh!! Thats why i was also stuck with the issue 😅😅.

Haha. Well one step at a time.

**Comment by subhradwip01** [comment]
> > Please don't print the list in the search component only. If you make the search component dynamic, we can use it on many pages like blog, resources, news etc. @kayceeDev
> 
> They're reasons this method won't work. You know the search component is also filtering the state and then rendering the filtered state. This means that it only filters the state it's given and end up Saving it in the components state. There is no way to pass the state of a child to the parent.
> 
> The other alternative might be to have a general state that the search component filters and this might require redux and some other complex logic. We can have a chat about this if you think there are easy ways to achieve this.


Maybe we can create a general search component that will take some props like data and a function from the blog component that function will just change the data inside the blog component on the basis of the search(filtered data that will be stored in the state of blog component) . In that way, we can use this search component in other components also.

**Comment by subhradwip01** [comment]
> > > Please don't print the list in the search component only. If you make the search component dynamic, we can use it on many pages like blog, resources, news etc. @kayceeDev
> > 
> > 
> > They're reasons this method won't work. You know the search component is also filtering the state and then rendering the filtered state. This means that it only filters the state it's given and end up Saving it in the components state. There is no way to pass the state of a child to the parent.
> > The other alternative might be to have a general state that the search component filters and this might require redux and some other complex logic. We can have a chat about this if you think there are easy ways to achieve this.
> 
> Maybe we can create a general search component that will take some props like data and a function from the blog component that function will just change the data inside the blog component on the basis of the search(filtered data that will be stored in the state of blog component) . In that way, we can use this search component in other components also.May be it sense

or we can create our own seach component without using js-seach package

**Comment by kayceeDev** [comment]
> > > Please don't print the list in the search component only. If you make the search component dynamic, we can use it on many pages like blog, resources, news etc. @kayceeDev
> > 
> > 
> > They're reasons this method won't work. You know the search component is also filtering the state and then rendering the filtered state. This means that it only filters the state it's given and end up Saving it in the components state. There is no way to pass the state of a child to the parent.
> > The other alternative might be to have a general state that the search component filters and this might require redux and some other complex logic. We can have a chat about this if you think there are easy ways to achieve this.
> 
> Maybe we can create a general search component that will take some props like data and a function from the blog component that function will just change the data inside the blog component on the basis of the search(filtered data that will be stored in the state of blog component) . In that way, we can use this search component in other components also.May be it sense

Thanks so much. You comments really gave me a nice intuition. I moved up the functions that performed the search in the parent component and passed on props that keeps track of the state change into the search component and it works. I will also see if I can create a custom hooks for the data searching. So that the custom hooks can actually be used whenever. I will see if this is possible. Nice job @subhradwip01

**Comment by subhradwip01** [comment]
> > > > Please don't print the list in the search component only. If you make the search component dynamic, we can use it on many pages like blog, resources, news etc. @kayceeDev
> > > 
> > > 
> > > They're reasons this method won't work. You know the search component is also filtering the state and then rendering the filtered state. This means that it only filters the state it's given and end up Saving it in the components state. There is no way to pass the state of a child to the parent.
> > > The other alternative might be to have a general state that the search component filters and this might require redux and some other complex logic. We can have a chat about this if you think there are easy ways to achieve this.
> > 
> > 
> > Maybe we can create a general search component that will take some props like data and a function from the blog component that function will just change the data inside the blog component on the basis of the search(filtered data that will be stored in the state of blog component) . In that way, we can use this search component in other components also.May be it sense
> 
> Thanks so much. You comments really gave me a nice intuition. I moved up the functions that performed the search in the parent component and passed on props that keeps track of the state change into the search component and it works. I will also see if I can create a custom hooks for the data searching. So that the custom hooks can actually be used whenever. I will see if this is possible. Nice job @subhradwip01
I am really happy to help you ☺️. You know I was also thinking to create a custome hook using js-search and use it in search component.

**Comment by subhradwip01** [comment]
> > > > Please don't print the list in the search component only. If you make the search component dynamic, we can use it on many pages like blog, resources, news etc. @kayceeDev
> > > 
> > > 
> > > They're reasons this method won't work. You know the search component is also filtering the state and then rendering the filtered state. This means that it only filters the state it's given and end up Saving it in the components state. There is no way to pass the state of a child to the parent.
> > > The other alternative might be to have a general state that the search component filters and this might require redux and some other complex logic. We can have a chat about this if you think there are easy ways to achieve this.
> > 
> > 
> > Maybe we can create a general search component that will take some props like data and a function from the blog component that function will just change the data inside the blog component on the basis of the search(filtered data that will be stored in the state of blog component) . In that way, we can use this search component in other components also.May be it sense
> 
> Thanks so much. You comments really gave me a nice intuition. I moved up the functions that performed the search in the parent component and passed on props that keeps track of the state change into the search component and it works. I will also see if I can create a custom hooks for the data searching. So that the custom hooks can actually be used whenever. I will see if this is possible. Nice job @subhradwip01
I am really happy to help you ☺️. You know I was also thinking to create a custome hook using js-search and use it in search component.

**Comment by l5io** [comment]
🚀 Preview for commit 860c232dad1ea5a72a0a927013d05671396ea407 at: https://6241e9f959850d0f3c4224da--layer5ng.netlify.app

**Comment by subhradwip01** [comment]
It is not working for the list view... also I guess rather than using it in the sidebar it should be in the grid and list component as we visit the mobile view it is gone to the bottom of the page...search bar should be at top of the page..

**Comment by l5io** [comment]
🚀 Preview for commit 7a0d8ebf410454b625985c7b1bb83bd1572365a3 at: https://6243583ac2a862007b7c2db9--layer5ng.netlify.app

**Comment by Nikhil-Ladha** [comment]
> > It is not working for the list view... also I guess rather than using it in the sidebar it should be in the grid and list component as we visit the mobile view it is gone to the bottom of the page...search bar should be at top of the page..
> 
> @Nikhil-Ladha said it is the desired implementation. That's why it's there. But I understand your point and will fix it to stay at the top.

Yes..but it seems like we need to use the design which you shared earlier on slack, i e, at the top right of blogs gird/list.

**Comment by kayceeDev** [comment]
> 🚀 Preview for commit [7a0d8eb](https://github.com/layer5io/layer5/commit/7a0d8ebf410454b625985c7b1bb83bd1572365a3) at: https://6243583ac2a862007b7c2db9--layer5ng.netlify.app

@Nikhil-Ladha this preview is the recent change with the search box at the top of the grid.

**Comment by Nikhil-Ladha** [comment]
> > rocket Preview for commit [7a0d8eb](https://github.com/layer5io/layer5/commit/7a0d8ebf410454b625985c7b1bb83bd1572365a3) at: https://6243583ac2a862007b7c2db9--layer5ng.netlify.app
> 
> @Nikhil-Ladha this preview is the recent change with the search box at the top of the grid.

Ok
But, I can't seem to understand on what basis is the search happening?

**Comment by kayceeDev** [comment]
> > > rocket Preview for commit [7a0d8eb](https://github.com/layer5io/layer5/commit/7a0d8ebf410454b625985c7b1bb83bd1572365a3) at: https://6243583ac2a862007b7c2db9--layer5ng.netlify.app
> > 
> > 
> > @Nikhil-Ladha this preview is the recent change with the search box at the top of the grid.
> 
> Ok But, I can't seem to understand on what basis is the search happening?

It searches the rendered blogs card by title and returns a match. So if a blog has the search input in its title? It renders only the matched cards. If you need more context, you can elaborate please.

**Comment by Nikhil-Ladha** [comment]
> > > > rocket Preview for commit [7a0d8eb](https://github.com/layer5io/layer5/commit/7a0d8ebf410454b625985c7b1bb83bd1572365a3) at: https://6243583ac2a862007b7c2db9--layer5ng.netlify.app
> > > 
> > > 
> > > @Nikhil-Ladha this preview is the recent change with the search box at the top of the grid.
> > 
> > 
> > Ok But, I can't seem to understand on what basis is the search happening?
> 
> It searches the rendered blogs card by title and returns a match. So if a blog has the search input in its title? It renders only the matched cards. If you need more context, you can elaborate please.

Sure.
Issues that I noticed:
1. We get only 1 result (which is not true).
2. If the search term enter doesn't match any data, it shows the complete list (instead it should show a text like "Sorry no match es" or something better)
3. The search doesn't happen on the complete data, and only on the blogs which are on that page.

The above are some initial findings, more can be discussed on Monday's call (if possible do fix above before that 😅 )

**Comment by kayceeDev** [comment]
> > > > > rocket Preview for commit [7a0d8eb](https://github.com/layer5io/layer5/commit/7a0d8ebf410454b625985c7b1bb83bd1572365a3) at: https://6243583ac2a862007b7c2db9--layer5ng.netlify.app
> > > > 
> > > > 
> > > > @Nikhil-Ladha this preview is the recent change with the search box at the top of the grid.
> > > 
> > > 
> > > Ok But, I can't seem to understand on what basis is the search happening?
> > 
> > 
> > It searches the rendered blogs card by title and returns a match. So if a blog has the search input in its title? It renders only the matched cards. If you need more context, you can elaborate please.
> 
> Sure. Issues that I noticed:
> 
> 1. We get only 1 result (which is not true).
> 2. If the search term entered doesn't match any data, it shows the complete list (instead it should show a text like "Sorry no matches" or something better)
> 3. The search doesn't happen on the complete data, and only on the blogs which are on that page.
> 
> The above are some initial findings, more can be discussed on Monday's call (if possible do fix the above before that 😅 )

For the first observations, I'd look into it possibly before the next call. Thanks very much.

**Comment by kayceeDev** [comment]
> > > > > rocket Preview for commit [7a0d8eb](https://github.com/layer5io/layer5/commit/7a0d8ebf410454b625985c7b1bb83bd1572365a3) at: https://6243583ac2a862007b7c2db9--layer5ng.netlify.app
> > > > 
> > > > 
> > > > @Nikhil-Ladha this preview is the recent change with the search box at the top of the grid.
> > > 
> > > 
> > > Ok But, I can't seem to understand on what basis is the search happening?
> > 
> > 
> > It searches the rendered blogs card by title and returns a match. So if a blog has the search input in its title? It renders only the matched cards. If you need more context, you can elaborate please.
> 
> Sure. Issues that I noticed:
> 
> 1. We get only 1 result (which is not true).
> 2. If the search term entered doesn't match any data, it shows the complete list (instead it should show a text like "Sorry no matches" or something better)
> 3. The search doesn't happen on the complete data, and only on the blogs which are on that page.
> 
> The above are some initial findings, more can be discussed on Monday's call (if possible do fix the above before that sweat_smile )

So on your observations, I have made changes now and the search no longer returns just one result but all the matches.
I will create something that renders when a search is not found.
for the last part, the data on the page is paginated and the search only works in the page context. What I mean here is that the graphQL query returns only the requested data when a user toggles to the next page and feeds it into the blog component and this is the data that the search box only sees and filters when a user searches for a blog.

It will be great if there is a way to query the data all at once and index it with the search plugin so that the search value goes through the whole data and not just the only data fed into the component on each page toggle. Pardon me if my explanations seems verbose

**Comment by leecalcote** [comment]
I am so very much looking forward to this functionality.

**Comment by shivaylamba** [comment]
My suggestion is to use a more powerful open source search engine like https://meilisearch.com. Would like to know your thoughts @leecalcote @Nikhil-Ladha

**Comment by kayceeDev** [comment]
> My suggestion is to use a more powerful open source search engine like https://meilisearch.com. Would like to know your thoughts @leecalcote @Nikhil-Ladha

The search plugin is not really the problem. I will discuss it in the next website meeting

**Comment by subhradwip01** [comment]
> > My suggestion is to use a more powerful open source search engine like https://meilisearch.com. Would like to know your thoughts @leecalcote @Nikhil-Ladha
> 
> The search plugin is not really the problem. I will discuss it in the next website meeting

Yes, lets discuss on next website call for more clarity

**Comment by l5io** [comment]
🚀 Preview for commit 803f9516c300a17594d05f942163402a959a667a at: https://6246f594c474552a7c7f155e--layer5ng.netlify.app

**Comment by Nikhil-Ladha** [comment]
Hey, @kayceeDev any updates from the Website's call? What's the next step?

**Comment by kayceeDev** [comment]
> Hey, @kayceeDev any updates from the Website's call? What's the next step?

Hi @Nikhil-Ladha, since the graphql query is getting data in 10s, and passing it onto the page context, that's why the search doesn't index the whole data. So it was suggested that I should adjust the query to get the all the data, so it can all be indexed by the search plugin. Also, @shivaylamba pointed me to other search plugins that might be more efficient. I intend to incorporate all the feedback and see if it works as intended.

**Comment by l5io** [comment]
🚀 Preview for commit 9cc08b42f83734d955c7e05b409fc00d353b6df1 at: https://6254b02bc23e833bb0f37d57--layer5ng.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 153b57d482673c249765a68f9c56110aaebecd57 at: https://6254bce863f1a1481fc44948--layer5ng.netlify.app

**Comment by Nikhil-Ladha** [comment]
Few issues:
1. Paging is not working if the search results expand multi-page.
2. Paging for any of the `Tags`, `Categories` is not working.
3. Why have you increased the number of items per page to 12, 14 (in search results)? Please revert it to 10.

**Comment by kayceeDev** [comment]
This was also discussed in the last meeting. I will update as required and update PR. Also, if merge can be considered for this PR that will be fine so I will create different PRs for all the updates above so that the PR won't get long.

**Comment by Nikhil-Ladha** [comment]
> This was also discussed in the last meeting. I will update as required and update PR. Also, if merge can be considered for this PR that will be fine so I will create different PRs for all the updates above so that the PR won't get long.

You can update this PR itself, no issues. I have gone through the code once everything seems fine. Not much trouble, will go again when the functionality is working as expected.
Some basic functionalities are not working as expected, so I would prefer to fix that before a merge.

**Comment by kayceeDev** [comment]
Alright.

**Comment by debo19** [comment]
@kayceeDev Can we discuss your updates on the call today?

**Comment by l5io** [comment]
🚀 Preview for commit 5daa6efbcb2e2c0f1203b5e7bf295d907bb5d9b7 at: https://6271420fbf80591215945306--layer5ng.netlify.app

File: src/reusecore/Search/index.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { FaSearch } from "@react-icons/all-files/fa/FaSearch";
+import React from "react";
+import { Search } from "./searchbox.style";
```
Comment (Nikhil-Ladha) [inline src/reusecore/Search/index.js:3]: Better name it `SearchWrapper` to align with the codebase naming structure.

Comment (kayceeDev) [reply src/reusecore/Search/index.js:3]: This is understandably an issue I am having @subhradwip01 too. I don't know why this is happening right now giving the fact the graphQL  query is working as it should. I will continue looking into it.

**Comment by l5io** [comment]
🚀 Preview for commit 2d6d9f65b4a9f6cde85a62a85d139a74fcfd10b0 at: https://62717784dc5a790b82a0842e--layer5ng.netlify.app

File: src/reusecore/Search/index.js (hunk 1 above)
Comment (kayceeDev) [reply src/reusecore/Search/index.js]: Everything now works with the latest commit.

**Comment by l5io** [comment]
🚀 Preview for commit 85e008a719d70e65bab84ca92f13a41dbd367569 at: https://62717ee5dc5a7912aca08ada--layer5ng.netlify.app

**Comment by subhradwip01** [comment]
LGTM😃

**Comment by leecalcote** [comment]
Say it ain't so... is Search ready to go?!? :)

**Comment by leecalcote** [comment]
Search isn't supported on /resources, yet, huh?

We don't give a count of the number of matching articles, no?

We don't have a site-wide search yet, eh?

**Comment by subhradwip01** [comment]
> Search isn't supported on /resources, yet, huh?
> 
> We don't give a count of the number of matching articles, no?
> 
> We don't have a site-wide search yet, eh?

Hmm...good points....now we have a architecture for searching in blog page...we may use this for otger pages...let discuss those things on next call...

**Comment by kayceeDev** [comment]
> Search isn't supported on /resources, yet, huh?
> 
> We don't give a count of the number of matching articles, no?
> 
> We don't have a site-wide search yet, eh?


Search only works on the blog page for now. Adding the search to the resources page would be the same way it was added to the blog page but easier now since a lot of figuring out has been done.

Also, it won't be a site wide search (if I understand what you mean) since each page has different data coming in, but with the way I created the solution, the search feature can easily be added across any page that needs it.

**Comment by kayceeDev** [comment]
> Search isn't supported on /resources, yet, huh?
> 
> We don't give a count of the number of matching articles, no?
> 
> We don't have a site-wide search yet, eh?

Also, we can actually give a count of the matching articles in the UI just like the behavior that happens when you are searching in the category or tag page.

**Review by Nikhil-Ladha — APPROVED** [review]
Looks good.
Good job @kayceeDev 💯

**Comment by Nikhil-Ladha** [comment]
@leecalcote sometimes it may be hard to understand you ;), but in any way, I don't think you mind getting this merged and continuing improvements on top of it.
Though, I will wait for your ack before merging this.

File: src/templates/blog.js [hunk 2]
```diff
@@ -11,71 +11,79 @@ import Footer from "../sections/General/Footer";
...
-    nodes {
-      id
-      frontmatter {
-        title
-        date(formatString: "MMM Do, YYYY")
-        author
-        thumbnail {
-          childImageSharp {
-            gatsbyImageData(layout: FULL_WIDTH)
+export const query = graphql`
```
Comment (leecalcote) [inline src/templates/blog.js:17]: @kayceeDev with all the reformatting included, it's difficult to tell what, if anything, has materially changed in this file? Anything?

File: src/templates/blog-category-list.js [hunk 3]
```diff
@@ -10,52 +10,56 @@ import Footer from "../sections/General/Footer";
 
 import { GlobalStyle } from "../sections/app.style";
 import theme from "../theme/app/themeStyles";
```
Comment (leecalcote) [inline src/templates/blog-category-list.js:12]: @kayceeDev is there any change other than formatting in this file?

**Comment by leecalcote** [comment]
> @leecalcote sometimes it may be hard to understand you ;), but in any way, I don't think you mind getting this merged and continuing improvements on top of it.
> Though, I will wait for your ack before merging this.

Thank goodness that you do so well deciphering my messages, @Nikhil-Ladha. Yes, that is a great plan. 👍 

@kayceeDev, confirmation: js-search was chosen over alternatives, because.... ? If this is written down somewhere, please just point me to the discussion.

**Comment by leecalcote** [comment]
How easy is it to propagate support for search site-wide using `js-search` vs an alternative library?

**Comment by leecalcote** [comment]
In some respects, irrespective of the answers to my questions above, it would be good to recognize (and merge) the work that has been done here, so that it can have a chance to be used, while be evaluated in context of those questions.

**Review by leecalcote — APPROVED** [review]

File: src/templates/blog.js (hunk 2 above)
Comment (kayceeDev) [reply src/templates/blog.js:17]: The only thing I changed in graphql query was to remove the limit of data returned per page so I can index all the data and make all of it searchable when a user uses the search bar. Else, our search future would be limited to the page context.

**Comment by kayceeDev** [comment]
> Looks good. Good job @kayceeDev 💯

Thanks @Nikhil-Ladha. You were really helpful.

**Comment by kayceeDev** [comment]
> > @leecalcote sometimes it may be hard to understand you ;), but in any way, I don't think you mind getting this merged and continuing improvements on top of it.
> > Though, I will wait for your ack before merging this.
> 
> Thank goodness that you do so well deciphering my messages, @Nikhil-Ladha. Yes, that is a great plan. 👍
> 
> @kayceeDev, confirmation: js-search was chosen over alternatives, because.... ? If this is written down somewhere, please just point me to the discussion.

So basically, Js-search wasn't just the option on the table but it did it work perfectly as it should also, it is very lightweight as compared to the other options on the table and the docs is very understandable too. Other options can be looked at why we use this for now.

**Comment by kayceeDev** [comment]
> How easy is it to propagate support for search site-wide using `js-search` vs an alternative library?

So I am trying to understand when you say a site wide search. The site doesn't use a single query to get all it's data. Some of the data are different by page context. If the site was using tools like redux to manage state, maybe that might be possible to get a site-wide search but this can be further deliberated on so I can get a clear context of what you mean here.

File: src/templates/blog-category-list.js (hunk 3 above)
Comment (kayceeDev) [reply src/templates/blog-category-list.js:12]: Yes. I made changed to the query of `blog.js` file. `Blog-tags-list.js` and `blog-category-list.js.` Then the search feature was then used in just the `blog-grid.js` and the `blog-list.js`

**Comment by Nikhil-Ladha** [comment]
> @kayceeDev, confirmation: js-search was chosen over alternatives, because.... ? If this is written down somewhere, please just point me to the discussion.

js-search is very similar to the other available library `elasticlunr-js`. But, `js-search`, looks comparatively better because of some additional configurations needed for `elasticlunr-js`.

While there are other options (which include melisearch, as suggested by Shivay), all of them are third-party search engines. And, it requires them to be hosted on a separate instance, send data to it, get it indexed, and fetch it again and use it on our site. Now, comparatively these search engines are more powerful (and they should be considering the algorithm is running on a completely separate instance) but they involved additional cost, some complexity and are better for a bigger database (which we aim to achieve sometime in the future! And, maybe that could be could be explored at that time)

**Pull request merged**